### PR TITLE
call clear_settings

### PR DESCRIPTION
**Pull request by MichaelChirico**
Feel like I'm missing something here, shouldn't this have been caught earlier? 🤔

### CODE REVIEW

**Comment by AshesITR** [comment]
This chunk trips all builds.
Can you debug?

```r
library(lintr)
linter_name <- "assignment_linter"

withr::with_tempfile("tmp", {
  writeLines("a = 1", tmp)

  # linter column is just 'get'
  print(as.data.frame(lint(tmp, linters = get(linter_name)())))

  this_linter <- get(linter_name)()
  attr(this_linter, "name") <- linter_name
  # linter column is 'assignment_linter'
  print(as.data.frame(lint(tmp, linters = this_linter)))

  # more concise alternative: use eval(call(.))
  print(as.data.frame(lint(tmp, linters = eval(call(linter_name)))))
})
```

**Comment by MichaelChirico** [comment]
> This chunk trips all builds. Can you debug?
> 
> ```r
> library(lintr)
> linter_name <- "assignment_linter"
> 
> withr::with_tempfile("tmp", {
>   writeLines("a = 1", tmp)
> 
>   # linter column is just 'get'
>   print(as.data.frame(lint(tmp, linters = get(linter_name)())))
> 
>   this_linter <- get(linter_name)()
>   attr(this_linter, "name") <- linter_name
>   # linter column is 'assignment_linter'
>   print(as.data.frame(lint(tmp, linters = this_linter)))
> 
>   # more concise alternative: use eval(call(.))
>   print(as.data.frame(lint(tmp, linters = eval(call(linter_name)))))
> })
> ```

Finally got around to this :)

`clear_settings()` deletes `settings$encoding`, so `read_lines()` was trying `iconv(from = NULL)` hence the bug. Didn't look into why it only happens there.

File: R/lint.R [hunk 1]
```diff
@@ -58,7 +58,7 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
 
   if (isTRUE(parse_settings)) {
```
Comment (MichaelChirico) [inline R/lint.R:59]: I see, actually there's a non-trivial issue. We only run `read_settings()` here, but `get_lines()` above relies on some settings. Not immediately clear how to switch things around. If `encoding=` is the only issue, then we can just allow `encoding=NULL` to work in a few places. Reading more carefully...

**Comment by MichaelChirico** [comment]
Subsumed by #2180

**Pull request closed without merging**